pkg/api/v1/types.go

```diff
@@ -1312,6 +1312,15 @@ type NodeSpec struct {
 	Unschedulable bool `json:"unschedulable,omitempty" description:"disable pod scheduling on the node; see http://releases.k8s.io/HEAD/docs/admin/node.md#manual-node-administration"`
 }
 
+type KubeletPorts struct {
+	// Port on which Kubelet is listening
+	KubeletPort uint `json:"kubeletPort" description:"port on which Kubelet is listening"`
+	// Read only port open on Kubelet
+	KubeletReadOnlyPorty uint `json:"kubeletReadOnlyPort" description:"read-only port exposed by Kubelet"`
+	// Port on which cadvisor proxy is running
+	CadvisorProxyPort uint `json:"cadvisorProxyPort" description:"port opened by Kubelet on which cadvisor proxy is available"`
```

Comment:
`omitempty` here, as well, if that doesn't complicate the code that needs it significantly.